data-loader/core/src/main/java/com/scalar/db/dataloader/core/DataLoaderError.java

```diff
@@ -0,0 +1,292 @@
+  MISSING_SOURCE_FIELD(
+      Category.USER_ERROR,
+      "0035",
+      "The data mapping source field '%s' for table '%s' is missing in the json data record",
+      "",
+      ""),
+  CSV_DATA_MISMATCH(
+      Category.USER_ERROR, "0036", "The CSV row: %s does not match header: %s", "", ""),
+  JSON_CONTENT_START_ERROR(
+      Category.USER_ERROR, "0037", "Expected JSON file content to be an array", "", ""),
+  IMPORT_TARGET_MISSING(
+      Category.USER_ERROR,
+      "0038",
+      "Missing option: either '--namespace' and'--table' or '--control-file' options must be specified",
+      "",
+      ""),
+  MISSING_IMPORT_FILE(
+      Category.USER_ERROR,
+      "0039",
+      "The file '%s' specified by the argument '%s' does not exist",
+      "",
+      ""),
+  LOG_DIRECTORY_WRITE_ACCESS_DENIED(
+      Category.USER_ERROR, "0040", "Cannot write to the log directory: %s", "", ""),
+  LOG_DIRECTORY_CREATION_FAILED(
+      Category.USER_ERROR, "0041", "Failed to create the log directory: %s", "", ""),
+  INVALID_CONTROL_FILE(Category.USER_ERROR, "0042", "Failed to parse the control file: %s", "", ""),
+  DIRECTORY_WRITE_ACCESS(
+      Category.USER_ERROR,
+      "0043",
+      "No permission to create or write files in the directory: %s",
+      "",
+      ""),
+  DIRECTORY_CREATION_FAILED(
+      Category.USER_ERROR, "0044", "Failed to create the directory: %s", "", ""),
+  PATH_IS_NOT_A_DIRECTORY(
+      Category.USER_ERROR, "0045", "Path exists but is not a directory: %s", "", ""),
+  FILE_PATH_IS_BLANK(Category.USER_ERROR, "0046", "File path must not be blank", "", ""),
+  FILE_NOT_FOUND(Category.USER_ERROR, "0047", "File not found: %s", "", ""),
+  INVALID_DATE_TIME_FOR_COLUMN_VALUE(
+      Category.USER_ERROR,
+      "0048",
+      "Invalid date time value '%s' specified for column %s in table %s in namespace %s",
+      "",
+      ""),
+  NULL_OR_EMPTY_KEY_VALUE_INPUT(
+      Category.USER_ERROR, "0049", "Key-value cannot be null or empty", "", ""),
+  INVALID_KEY_VALUE_INPUT(Category.USER_ERROR, "0050", "Invalid key-value format: %s", "", ""),
+  SPLIT_INPUT_VALUE_NULL(Category.USER_ERROR, "0051", "Value must not be null", "", ""),
+  SPLIT_INPUT_DELIMITER_NULL(Category.USER_ERROR, "0052", "Delimiter must not be null", "", ""),
+  CONFIG_FILE_PATH_BLANK(Category.USER_ERROR, "0053", "Config file path must not be blank", "", ""),
+  INVALID_DATA_CHUNK_SIZE(
+      Category.USER_ERROR, "0054", "Data chunk size must be greater than 0", "", ""),
+  INVALID_TRANSACTION_SIZE(
+      Category.USER_ERROR, "0055", "Transaction size must be greater than 0", "", ""),
+  INVALID_MAX_THREADS(
+      Category.USER_ERROR, "0056", "Number of max threads must be greater than 0", "", ""),
+
+  //
+  // Errors for the internal error category
+  //
+  ERROR_CRUD_EXCEPTION(
+      Category.INTERNAL_ERROR,
+      "0000",
+      "Something went wrong while trying to save the data. Details: %s",
```

Comment:
I think `something went wrong` is a little too casual. How about the following instead?

```suggestion
      "A problem occurred while trying to save the data. Details: %s",
```